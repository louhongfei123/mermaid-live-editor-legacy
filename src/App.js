import React from 'react'
import { Row, Col, Input, Icon } from 'antd'

class App extends React.Component {
  render () {
    return (
      <Row gutter={16}>
        <Col span={6}>
          <Input.TextArea rows={6} value={'graph LR\nA-->B'} />
        </Col>
        <Col span={18}>
          <h1>Hello mermaid live editor!</h1>
          <h2><Icon type='pushpin' /> Work in progress...</h2>
        </Col>
      </Row>
    )
  }
}

export default App
