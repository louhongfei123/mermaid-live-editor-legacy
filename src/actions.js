export const setProp = (path, value) => {
  if (!Array.isArray(path)) {
    path = [path]
  }
  return { type: 'SET_PROP', path, value }
}

export const loadState = () => ({ type: 'LOAD_STATE' })

export const setState = (state) => ({ type: 'SET_STATE', state })

export const renderMermaid = (value) => ({ type: 'RENDER_MERMAID', value })
