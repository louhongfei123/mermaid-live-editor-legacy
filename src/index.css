html, body {
  padding: 16px;
}

.separator {
  border-bottom: 1px dotted gray;
  margin: 16px 0;
}
