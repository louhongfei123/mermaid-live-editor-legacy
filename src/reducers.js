import R from 'ramda'

const defaultState = {
  value: ''
}

const reducer = (state = defaultState, action) => {
  switch (action.type) {
    case 'SET_PROP':
      return R.set(R.lensPath(action.path), action.value)
    case 'SET_STATE':
      return R.always(action.state)
    default:
      if (!R.contains(action.type, ['@@redux/INIT', 'LOAD_STATE', 'RENDER_MERMAID'])) {
        // When app starts, '@@redux/INIT' invoked by Redux
        // Others are handled by redux-logic
        console.error(`Unknown action type: ${action.type}`)
      }
      return R.always(state)
  }
}

export default (state, action) => reducer(state, action)(state)
