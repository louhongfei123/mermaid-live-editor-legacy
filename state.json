{
  "defaultValue": "sequenceDiagram\nA->> B: Query\nB->> C: Forward query\nNote right of C: Thinking...\nC->> B: Response\nB->> A: Forward response",
  "value": ""
}
