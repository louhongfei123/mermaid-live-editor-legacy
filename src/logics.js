import { createLogic } from 'redux-logic'
import axios from 'axios'

import { setState } from './actions'

const loadStateLogic = createLogic({
  type: 'LOAD_STATE',
  latest: true,
  async process ({ getState, action }, dispatch, done) {
    const res = await axios.get('/state.json')
    dispatch(setState(res.data))
    done()
  }
})

const renderMermaidLogic = createLogic({
  type: 'RENDER_MERMAID',
  debounce: 1000,
  latest: true,
  async process ({ getState, action }, dispatch, done) {
    console.log('render mermaid')
    const element = action.mermaidContainer
    element.removeAttribute('data-processed')
    let mermaidError = null
    window.mermaid.parseError = (error, hash) => {
      mermaidError = error
    }
    const value = getState().value
    if (window.mermaid.parse(value) || mermaidError === null) {
      element.innerHTML = value
      window.mermaid.init(undefined, element)
    } else {
      element.innerHTML = `<pre>${mermaidError}</pre>`
    }
    done()
  }
})

export default [
  loadStateLogic,
  renderMermaidLogic
]
