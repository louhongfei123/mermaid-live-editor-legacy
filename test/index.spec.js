/* eslint-env jest */
import { Input } from 'antd'

import App from '../src/App'
import { getWrapper, defaultValue, timeout } from './shared'
import state from '../state.json'
import store from './store'
import { setProp } from '../src/actions'

let wrapper = null
beforeEach(() => {
  wrapper = getWrapper(App, state)
})

test('load state', async () => {
  await timeout(1000)
  expect(store.getState().value).toEqual(defaultValue.value)
})

test('set property', () => {
  store.dispatch(setProp('value', 'hello'))
  expect(store.getState().value).toEqual('hello')
  store.dispatch(setProp(['value'], 'world')) // first param is either string or array
  expect(store.getState().value).toEqual('world')
})

test('edit value', () => {
  const textArea = wrapper.find(Input.TextArea).first()
  const value = 'Hello world'
  textArea.props().onChange({ target: { value } })
  expect(textArea.props().value).toEqual(value)
  expect(store.getState().value).toEqual(value)
})
