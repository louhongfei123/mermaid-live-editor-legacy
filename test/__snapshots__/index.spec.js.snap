// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit value 1`] = `
<Provider
  store={
    Object {
      "actions": Array [
        Object {
          "type": "LOAD_STATE",
        },
        Object {
          "path": Array [
            "value",
          ],
          "type": "SET_PROP",
          "value": "Hello world",
        },
      ],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      loadState={[Function]}
      setProp={[Function]}
      value="Hello world"
    >
      <Row
        gutter={16}
      >
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -8,
              "marginRight": -8,
            }
          }
        >
          <Col
            span={6}
            style={
              Object {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-col-6"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <TextArea
                onChange={[Function]}
                prefixCls="ant-input"
                rows={6}
                value="Hello world"
              >
                <textarea
                  className="ant-input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  rows={6}
                  style={Object {}}
                  value="Hello world"
                />
              </TextArea>
            </div>
          </Col>
          <Col
            span={18}
            style={
              Object {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-col-18"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <h1>
                Hello mermaid live editor!
              </h1>
              <pre>
                Hello world
              </pre>
            </div>
          </Col>
        </div>
      </Row>
    </App>
  </Connect(App)>
</Provider>
`;

exports[`load state 1`] = `
<Provider
  store={
    Object {
      "actions": Array [
        Object {
          "type": "LOAD_STATE",
        },
        Object {
          "state": Object {
            "value": "graph LR
    A-->B",
          },
          "type": "SET_STATE",
        },
      ],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      loadState={[Function]}
      setProp={[Function]}
      value="graph LR
    A-->B"
    >
      <Row
        gutter={16}
      >
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -8,
              "marginRight": -8,
            }
          }
        >
          <Col
            span={6}
            style={
              Object {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-col-6"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <TextArea
                onChange={[Function]}
                prefixCls="ant-input"
                rows={6}
                value="graph LR
              A-->B"
              >
                <textarea
                  className="ant-input"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  rows={6}
                  style={Object {}}
                  value="graph LR
                A-->B"
                />
              </TextArea>
            </div>
          </Col>
          <Col
            span={18}
            style={
              Object {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-col-18"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <h1>
                Hello mermaid live editor!
              </h1>
              <pre>
                graph LR
                A--&gt;B
              </pre>
            </div>
          </Col>
        </div>
      </Row>
    </App>
  </Connect(App)>
</Provider>
`;
